remoteworkitem/flatten.go

```diff
@@ -0,0 +1,57 @@
+package remoteworkitem
+
+import (
+	"reflect"
+	"strings"
+)
+
+type stack []string
+
+func (s *stack) push(v string) {
+	*s = append(*s, v)
+}
+
+func (s *stack) pop() string {
+	l := len(*s)
+	if l == 0 {
+		return ""
+	}
+	stackReference := *s // because *s[:l-1] did not work.
+	poppedData := stackReference[l-1]
+	*s = stackReference[:l-1]
+	return poppedData
+}
+
+func (s stack) String() string {
+	return strings.Join(s, ".")
+}
+
+// nestedKeyStack stores the stack of keys till a leaf node is found.
+var nestedKeyStack = make(stack, 0)
```

Comment:
**ATTENTION: thread-safety**

Having `nestedKeyStack` defined as a global variable to which other functions append or read from doesn't make this thread-safe. I suggest to create a struct and with a `nestedKeyStack` member and make the `traverseMapAsDFS` and `Flatten` functions methods of this struct.